src/app.py

```diff
@@ -13,39 +29,114 @@ def set_log_level(verbose):
+     
+# Import Routes
+from . import routes
+
+def run_flask_app():
+	flask_app.run(ssl_context='adhoc', debug=True, use_reloader=False)
+
+@click.group()
+def cli():
+    pass
+
+@cli.command()
+@click.argument('token_id', required=False)
+def tokens(token_id):
+    """Show tokens. If TOKEN_ID is provided, show the full token for that ID."""
+    show_tokens(token_id)
+
+def show_tokens(token_id=None):
+    try:
+        with open('tokens.txt', 'r') as file:
+            tokens = file.readlines()
+
+        if not tokens:
+            print("No tokens found.")
+            return
+
+        if token_id is not None:
+            # Display the full token for the given ID
+            for line in tokens:
+                parts = line.strip().split(',')
+                if len(parts) == 2 and parts[0] == str(token_id):
+                    _, full_token = parts
+                    print(f"Full token for ID {token_id}: {full_token}")
+                    return
+            print(f"No token found with ID {token_id}")
+        else:
+            # Print table header
+            print(f"{'ID':<5} | {'Token':<10} | {'Date':<10} | {'Time':<10}")
+            print("-" * 50)  # Adjust the length based on your table width
+
+            # Print each token in brief
+            for line in tokens:
+                parts = line.strip().split(',')
+                if len(parts) == 2:
+                    id, token = parts
+                    token_preview = token[:7] + "..."
+                    print(f"{id:<5} | {token_preview:<10}")
+                else:
+                    print("Invalid line format:", line)
+
+    except FileNotFoundError:
+        print("No tokens found.")
+
+@cli.command()
+def url():
+    """Print and log the constructed URL."""
+    if all([global_endpoint, global_client_id, global_scope, global_response_type, global_redirect_uri]):
+        constructed_url = (f'{global_endpoint}?client_id={global_client_id}&scope={global_scope}&'
+                           f'response_type={global_response_type}&redirect_uri={global_redirect_uri}')
+        print(constructed_url)
+    else:
+        print("URL components are not fully set.")
+
+def command_loop():
+    history = InMemoryHistory()
+    token_completer = WordCompleter(['tokens', 'url', 'exit'], ignore_case=True)
+    while True:
+        try:
+            cmd = prompt("Enter command: ", completer=token_completer, style=custom_style, history=history)
+            parts = cmd.split()
+            if parts:
+                if parts[0] in ['tokens', 'url']:
+                    cli.main(args=parts, standalone_mode=False)
+                elif parts[0] == "exit":
+                    break
+                else:
+                    print("Unknown command.")
+        except Exception as e:
+            print(f"An error occurred: {e}")
 
 if __name__ == '__main__':
-	main()
+    main()
+    flask_thread = threading.Thread(target=run_flask_app)
```

Comment:
Switch to async/await architecture. Not threading.

Flask already runs in its own threads. Let flask worry about threading.